### ISSUE TITLE
Add integration test for --cache-images for all container runtimes

### ISSUE THREAD

**Issue by tstromberg** [kubernetes/minikube#3651]
--cache-images is in bad shape right now.

**Comment by afbjorklund** [comment]
Yeah, we need a new ISO for CRI-O and the mutex fix (#3643) for everyone else.

Also, it seems to be very happy about downloading duplicate and/or older images.

**Comment by afbjorklund** [comment]
Example of duplicates:
``` console
$ sudo crictl images
IMAGE                                      TAG                 IMAGE ID            SIZE
gcr.io/k8s-minikube/storage-provisioner    v1.8.1              4689081edb103       80.8MB
k8s.gcr.io/coredns                         1.1.3               b3b94275d97cb       46.2MB
k8s.gcr.io/coredns                         1.2.2               367cdc8433a45       39.5MB
k8s.gcr.io/coredns                         1.2.6               f59dcacceff45       40.3MB
k8s.gcr.io/etcd-amd64                      3.2.18              b8df3b177be23       219MB
k8s.gcr.io/etcd                            3.2.24              3cab8e1b9802c       220MB
k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64     1.14.8              c2ce1ffb51ed6       41.2MB
k8s.gcr.io/k8s-dns-kube-dns-amd64          1.14.8              80cc5ea4b547a       50.7MB
k8s.gcr.io/k8s-dns-sidecar-amd64           1.14.8              6f7f2dc7fab5d       42.5MB
k8s.gcr.io/kube-addon-manager              v8.6                9c16409588eb1       80.6MB
k8s.gcr.io/kube-apiserver                  v1.13.3             fe242e556a995       183MB
k8s.gcr.io/kube-apiserver-amd64            v1.13.3             fe242e556a995       183MB
k8s.gcr.io/kube-controller-manager         v1.13.3             0482f64009330       148MB
k8s.gcr.io/kube-controller-manager-amd64   v1.13.3             0482f64009330       148MB
k8s.gcr.io/kube-proxy                      v1.13.3             98db19758ad48       82.1MB
k8s.gcr.io/kube-proxy-amd64                v1.13.3             98db19758ad48       82.1MB
k8s.gcr.io/kube-scheduler                  v1.13.3             3a6f709e97a00       81.2MB
k8s.gcr.io/kube-scheduler-amd64            v1.13.3             3a6f709e97a00       81.2MB
k8s.gcr.io/kubernetes-dashboard-amd64      v1.10.1             f9aed6605b814       122MB
k8s.gcr.io/pause-amd64                     3.1                 da86e6ba6ca19       746kB
k8s.gcr.io/pause                           3.1                 da86e6ba6ca19       746kB
```

Some of these cache images are quite costly, such as etcd (220M)
